src/Pagination/PaginateDirective.php

```diff
@@ -126,18 +126,33 @@ public function resolveField(FieldValue $fieldValue): FieldValue
                     $this->directiveArgValue('scopes') ?? []
                 );
 
-            return PaginationArgs::extractArgs($args, $this->paginationType(), $this->paginateMaxCount())
+            return PaginationArgs::extractArgs($args, $this->paginationType($resolveInfo), $this->paginateMaxCount())
                 ->applyToBuilder($query);
         });
 
         return $fieldValue;
     }
 
-    protected function paginationType(): PaginationType
+    protected function paginationType(?ResolveInfo $resolveInfo = null): PaginationType
     {
-        return new PaginationType(
+        $type = new PaginationType(
             $this->directiveArgValue('type') ?? PaginationType::PAGINATOR
         );
+
+        if (null === $resolveInfo || $type->isSimple()) {
+            return $type;
+        }
+
+        $pageInfoFieldName = $type->isPaginator()
+            ? 'paginatorInfo'
+            : 'pageInfo';
```

Comment:
Those should use constants that are also used in code generation to make the relation clear.